### ISSUE TITLE
TopoJSON driver not supported since version 1.8a3

### ISSUE THREAD

**Issue by mattijn** [Toblerity/Fiona#721]
Until version `1.8a2` the following code was working (although with a new requirement of `boto3`). 
But since version `1.8a3` there is raised an error stating: 

`DriverError: unsupported driver: 'TopoJSON'`:

```python
import fiona
import geopandas as gpd
import requests

# parse the topojson file into memory
request = requests.get('https://vega.github.io/vega-datasets/data/us-10m.json')
visz = fiona.ogrext.buffer_to_virtual_file(bytes(request.content))

# read the features from a fiona collection into a GeoDataFrame
with fiona.Collection(visz, driver='TopoJSON') as f:
    gdf = gpd.GeoDataFrame.from_features(f, crs=f.crs)
```

I can't find the specific commit, but is it related to this new `fiona.Env()` function? How would/should this work?

**Comment by snorfalorpagus** [comment]
@mattijn Which operating system are you using? How did you install Fiona (from source, wheel, conda)?

**Comment by mattijn** [comment]
I'm on a Mac (OS X 10.14.3), Python 3.7.2 and installed Fiona through PyPI. 
This specific version I installed as such:
```python3 -m pip install fiona==1.8a3```

**Comment by sgillies** [comment]
@mattijn can you test whether the problem exists in Fiona 1.8.4? There's nothing we can do about 1.8a3, which was a prerelease and is now obsolete.

**Comment by mattijn** [comment]
Yes, the problem occurs since 1.8a3. With 1.8.4 I got the same `DriverError`.



```python
import fiona
import geopandas as gpd
import requests
```


```python
fiona.__version__
```




    '1.8.4'




```python
# parse the topojson file into memory
request = requests.get('https://vega.github.io/vega-datasets/data/us-10m.json')
visz = fiona.ogrext.buffer_to_virtual_file(bytes(request.content))

# read the features from a fiona collection into a GeoDataFrame
with fiona.Collection(visz, driver='TopoJSON') as f:
    gdf = gpd.GeoDataFrame.from_features(f, crs=f.crs)
```


    ---------------------------------------------------------------------------

    DriverError                               Traceback (most recent call last)

    <ipython-input-3-0436d13e1a80> in <module>()
          4 
          5 # read the features from a fiona collection into a GeoDataFrame
    ----> 6 with fiona.Collection(visz, driver='TopoJSON') as f:
          7     gdf = gpd.GeoDataFrame.from_features(f, crs=f.crs)


    /usr/local/lib/python3.7/site-packages/fiona/collection.py in __init__(self, path, mode, driver, schema, crs, encoding, layer, vsi, archive, enabled_drivers, crs_wkt, ignore_fields, ignore_geometry, **kwargs)
        166 
        167         if self.session is not None:
    --> 168             self.guard_driver_mode()
        169             if not self.encoding:
        170                 self.encoding = self.session.get_fileencoding().lower()


    /usr/local/lib/python3.7/site-packages/fiona/collection.py in guard_driver_mode(self)
        185         driver = self.session.get_driver()
        186         if driver not in supported_drivers:
    --> 187             raise DriverError("unsupported driver: %r" % driver)
        188         if self.mode not in supported_drivers[driver]:
        189             raise DriverError("unsupported mode: %r" % self.mode)


    DriverError: unsupported driver: 'TopoJSON'



```python

```


Maybe there is a recent mapping in Fiona somewhere to only allow OGR vector formats that allow creation? As the TopoJSON format is only supported for reading in OGR, https://www.gdal.org/ogr_formats.html.

**Comment by mattijn** [comment]
OGR seems to have no problems with the file if I try to read it the info from command line:
```bash
ogrinfo -ro "https://vega.github.io/vega-datasets/data/us-10m.json"
```

```bash
INFO: Open of `https://vega.github.io/vega-datasets/data/us-10m.json'
      using driver `TopoJSON' successful.
1: counties
2: states
3: TopoJSON (Multi Polygon)
```

**Comment by sgillies** [comment]
Thanks @mattijn . There is indeed a bug: I overlooked that TopoJSON is no longer handled by the GeoJSON driver beginning in GDAL 2.3.0 and have not yet added it to the registry at https://github.com/Toblerity/Fiona/blob/master/fiona/drvsupport.py#L13.

```
>>> import requests
>>> from fiona.io import MemoryFile
>>> resp = requests.get('https://vega.github.io/vega-datasets/data/us-10m.json')
>>> with MemoryFile(resp.content) as memfile:
...     with memfile.open(driver='TopoJSON') as collection:
...         print(collection.meta)
...         print(len(collection))
...
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "/Users/seang/envs/data/lib/python3.6/site-packages/fiona/io.py", line 55, in open
    **kwargs)
  File "/Users/seang/envs/data/lib/python3.6/site-packages/fiona/collection.py", line 168, in __init__
    self.guard_driver_mode()
  File "/Users/seang/envs/data/lib/python3.6/site-packages/fiona/collection.py", line 187, in guard_driver_mode
    raise DriverError("unsupported driver: %r" % driver)
fiona.errors.DriverError: unsupported driver: 'TopoJSON'
```

The Fiona 1.8a3 wheels that I put on PyPI were the first to include GDAL 2.3. This issue will be fixed in Fiona 1.8.5.

**Comment by mattijn** [comment]
Thanks for the quick fix!
